### PR TITLE
[workers] remove references to deprecated commands

### PR DESCRIPTION
**Pull request by admah**
This PR:

- removes references to: `--persist`, `--local`, `--experimental-local`
- updates references for `wrangler publish` to `wrangler deploy`
- remove warnings related to non-default local mode that don't apply to v3

### CODE REVIEW

**Review by rozenmd — APPROVED** [review]
PR looks good, do we have a plan on documenting how to do things in v2.x?

Might be worth mentioning --persist and --local and --experimental-local in https://developers.cloudflare.com/workers/wrangler/deprecations/ ?

**Comment by WalshyDev** [comment]
can we revert all the auto-formatting? (in vscode you can do "Save without formatting" to stop it doing this)

File: content/workers/wrangler/deprecations.md [hunk 1]
```diff
@@ -26,18 +26,32 @@ The `wrangler init` command is deprecated, but still active in v3. It will be fu
...
-### `pages publish`
+#### `pages publish`
 
 The `wrangler pages publish` command is deprecated, but still active in v3. `wrangler pages publish` will be fully removed in v4.
 
 Try using [`wrangler pages deploy`](/workers/wrangler/commands/#deploy-1) to deploy Pages.
 
+### Deprecated options
+
+### `--experimental-local`
```
Comment (rozenmd) [inline content/workers/wrangler/deprecations.md]:
```suggestion
#### `--experimental-local`
```

File: content/workers/wrangler/deprecations.md [hunk 2]
```diff
@@ -26,18 +26,32 @@ The `wrangler init` command is deprecated, but still active in v3. It will be fu
...
+
+`wrangler dev` in v3 is local by default so this option is no longer necessary.
+
+### `--local`
```
Comment (rozenmd) [inline content/workers/wrangler/deprecations.md]:
```suggestion
#### `--local`
```

File: content/workers/wrangler/deprecations.md [hunk 3]
```diff
@@ -26,18 +26,32 @@ The `wrangler init` command is deprecated, but still active in v3. It will be fu
...
+
+`wrangler dev` in v3 is local by default so this option is no longer necessary.
+
+### `--persist`
```
Comment (rozenmd) [inline content/workers/wrangler/deprecations.md]:
```suggestion
#### `--persist`
```

**Review by deadlypants1973 — APPROVED** [review]

**Review by elithrar — APPROVED** [review]

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**